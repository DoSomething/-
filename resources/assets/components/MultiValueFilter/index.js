import React from 'react';
import { map } from 'lodash';
import classnames from 'classnames';

import '../Tags/tags.scss';

class MultiValueFilter extends React.Component {
  constructor() {
    super();

    this.state = {};

    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    const type = this.props.options.type;
    const values = this.props.options.values;
    const options = {
      [type]: values,
    };
    this.setState({ ...options });
  }

  componentDidUpdate() {
    this.props.updateFilters(this.state);
  }

  shouldComponentUpdate(nextProps, nextState) {
    return nextState !== this.state;
  }

  handleClick(key, activeFilter, type) {
    const values = {
      active: !activeFilter,
      label: this.state[type][key].label,
    };

    this.setState((previousState) => {
      const newState = {...previousState};
      newState[type][key] = values;

      return newState;
    });
  }

  render() {
    return (
      <div className="container__block -third">
        <h2 className="heading -delta">{this.props.header}</h2>
        <ul className="aligned-actions">
          {map(Object.values(this.state)[0], (option, key) => (
            <li key={key}>
              {Object.values(this.state)[0] ?
                <button className={classnames('tag', {'is-active':  option.active})}
                      onClick={() => this.handleClick(key, option.active, this.props.options.type)}>{option.label}</button>
              : null}
            </li>
          ))}
        </ul>
      </div>
    )
  }
}

export default MultiValueFilter;
