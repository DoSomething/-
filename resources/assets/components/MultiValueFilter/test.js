import React from 'react';
import { handleClick } from '../MultiValueFilter';

test('handleClick changes button to active', () => {
  const startState = {
    tags: {
      'good-photo': {
        active: false,
        label: 'Good Photo',
      }
    }
  };
  const finState = handleClick('good-photo', false, 'tags');

  expect(finState.tags['good-photo']).toEqual([
    { active: true, label: 'Good Photo' }
  ]);
});
