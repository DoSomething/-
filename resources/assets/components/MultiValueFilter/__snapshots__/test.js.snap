// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a list of tags 1`] = `
<MultiValueFilter
  header="Tags"
  options={
    Object {
      "type": "tags",
      "values": Object {
        "good-quote": Object {
          "active": false,
          "label": "Good Quote",
        },
        "good-submission": Object {
          "active": false,
          "label": "Good Submission",
        },
      },
    }
  }
  updateFilters={[Function]}
>
  <div
    className="container__block -third"
  >
    <h2
      className="heading -delta"
    >
      Tags
    </h2>
    <ul
      className="aligned-actions"
    >
      <li
        key="good-submission"
      >
        <button
          className="tag"
          onClick={[Function]}
        >
          Good Submission
        </button>
      </li>
      <li
        key="good-quote"
      >
        <button
          className="tag"
          onClick={[Function]}
        >
          Good Quote
        </button>
      </li>
    </ul>
  </div>
</MultiValueFilter>
`;
